addons/l10n_it/models/l10n_it_account_invoice.py

```diff
@@ -0,0 +1,421 @@
+                raise UserError(_("%s isn't in a right state. It must be in a 'Not yet send' or 'Invalid' state.") % (invoice.display_name))
+
+        if not self.company_id.mail_pec_server_id or not self.company_id.address_send_fatturapa:
+            for invoice in self:
+                invoice.message_post(
+                    body=(_("Error when sending mail with E-Invoice: Your company must have a mail PEC server and must indicate the mail PEC that will send electronic invoice."))
+                    )
+                invoice.send_state = ('invalid')
+            return
+
+        attachment = self.env['ir.attachment'].search([
+            ('res_id', 'in', self.ids),
+            ('res_model', '=', self._name),
+            ('mimetype', 'ilike', 'xml')
+        ])
+
+        message = self.env['mail.message'].create({
+            'subject': _('Sending file: %s') % (' '.join(attachment.mapped('name'))),
+            'body': _('Sending file: %s to ES: %s') % (' '.join(attachment.mapped('name')), self.env.user.company_id.address_recipient_fatturapa),
+            'email_from': self.env.user.company_id.address_send_fatturapa,
+            'mail_server_id': self.env.user.company_id.mail_pec_server_id.id,
+            'attachment_ids': [(6, 0, attachment.ids)],
+        })
+
+        mail_fattura = self.env['mail.mail'].create({
+            'mail_message_id': message.id,
+            'email_to': self.env.user.company_id.address_recipient_fatturapa,
+        })
+        try:
+            mail_fattura.send(raise_exception=True)
+            for invoice in self:
+                invoice.message_post(
+                    body=(_("Mail sent on %s by %s") % (fields.Datetime.now(), self.env.user.display_name))
+                    )
+                invoice.send_state = ('sent')
+        except MailDeliveryException as error:
+            for invoice in self:
+                invoice.message_post(
+                    body=(_("Error when sending mail with E-Invoice: %s") % (error.args[0]))
+                    )
+                invoice.send_state = ('invalid')
+
+class AccountInvoiceLine(models.Model):
+    _inherit = 'account.invoice.line'
+    _name = 'account.invoice.line'
+
+    ddt_line_ids = fields.Many2many('l10n.it.ddt', string='DDT', copy=True)
+
+class AccountTax(models.Model):
+    _name = "account.tax"
+    _inherit = "account.tax"
+
+    VAT_due_date = fields.Selection([
+        ("I", "[I] IVA ad esigibilità immediata"),
+        ("D", "[D] IVA ad esigibilità differita"),
+        ("S", "[S] Scissione dei pagamenti")], default="I", string="VAT due date")
+
+    has_exoneration = fields.Boolean(string="Has exoneration of tax", help="Tax has a tax exoneration.")
+    kind_exoneration = fields.Char(string="Exoneration", size=2, help="Exoneration type: First letter must be a N, second must be a integer. For example: N1 or N6", default="N1")
+    law_reference = fields.Char(string="Law Reference", size=100, default="Law ...")
```

Comment:
Maybe better to use a placeholder in the view than a default value that makes no sense.